packages/teams-js/src/private/externalAppAuthentication.ts

```diff
@@ -0,0 +1,329 @@
+import { sendMessageToParentAsync } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { authentication, FrameContexts } from '../public';
+import { errorNotSupportedOnPlatform } from '../public/constants';
+import { runtime } from '../public/runtime';
+
+/**
+ * @hidden
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+export namespace externalAppAuthentication {
+  /*********** BEGIN REQUEST TYPE ************/
+  /**
+   * @hidden
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export type OriginalRequestInfo = IQueryMessageExtensionRequest | IActionExecuteInvokeRequest;
+
+  /**
+   * @hidden
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export interface IQueryMessageExtensionRequest {
+    requestType: OriginalRequestType.QueryMessageExtensionRequest;
+    commandId: string;
+    parameters?: {
+      name?: string;
+      value?: string;
+    }[];
+    queryOptions?: {
+      count: number;
+      skip: number;
+    };
+  }
+
+  /**
+   * @hidden
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  interface IActionExecuteInvokeRequest {
+    requestType: OriginalRequestType.ActionExecuteInvokeRequest;
+    type: string; // "invoke"
+    id: string; // "action id"
+    verb: string; // "action"
+    data: Record<string, unknown>; //object; // {}
+  }
+
+  /**
+   * @hidden
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export enum OriginalRequestType {
+    ActionExecuteInvokeRequest = 'ActionExecuteInvokeRequest',
+    QueryMessageExtensionRequest = 'QueryMessageExtensionRequest',
+  }
+  /*********** END REQUEST TYPE ************/
+
+  /*********** BEGIN RESPONSE TYPE ************/
+  /**
+   * @hidden
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export type IInvokeResponse = IQueryMessageExtensionResponse | IActionExecuteResponse;
+
+  /**
+   * @hidden
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export enum InvokeResponseType {
+    ActionExecuteInvokeResponse = 'ActionExecuteInvokeResponse',
+    QueryMessageExtensionResponse = 'QueryMessageExtensionResponse',
+  }
+
+  /**
+   * @hidden
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export interface IQueryMessageExtensionResponse {
+    responseType: InvokeResponseType.QueryMessageExtensionResponse;
+    composeExtension?: ComposeExtensionResponse;
+  }
+
+  /**
+   * @hidden
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export interface IActionExecuteResponse {
+    responseType: InvokeResponseType.ActionExecuteInvokeResponse;
+    value: Record<string, unknown>;
+    signature?: string;
+    statusCode: number;
+    type: string;
+  }
+
+  /**
+   * @hidden
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export type ComposeExtensionResponse = {
```

Comment:
>type [](http://example.com/codeflow?start=9&length=4)

Why did you choose to use `type` in some places and `interface` in others? Did you intend to leverage the difference between these (that you can't extend `type` elsewhere)?

Reply:
@ashishsingh2105 did you decide on most of these distinctions? Are the type/interface objects intentionally different?